src/vm_stack.zig

```diff
@@ -5303,24 +5338,35 @@ pub const StackVM = struct {
         }
     }
 
-    pub fn resumeInvoke(vm: *VM, module: *ModuleInstance, returns: []Val) anyerror!void {
+    pub fn resumeInvoke(vm: *VM, module: *ModuleInstance, returns: []Val, opts: ResumeInvokeOpts) anyerror!void {
         var self: *StackVM = fromVM(vm);
 
-        std.debug.assert(self.debug_state != null);
-        std.debug.assert(self.debug_state.?.is_invoking);
-
-        const debug_state = &self.debug_state.?;
-        const opcode: Opcode = blk: {
+        var pc: u32 = 0;
+        var opcode = Opcode.Invalid;
+        if (self.debug_state) |debug_state| {
+            std.debug.assert(debug_state.is_invoking);
+            pc = debug_state.pc;
             for (debug_state.trapped_opcodes.items) |op| {
                 if (op.address == debug_state.pc) {
-                    break :blk op.opcode;
+                    opcode = op.opcode;
+                    break;
                 }
             }
             unreachable; // Should never get into a state where a trapped opcode doesn't have an associated record
-        };
+
+        } else {
```

Comment:
should this be `else if (metering.enabled)`?

Reply:
Yeah, that makes it cleaner. I've added an else clause too so that we have an error case.